rskj-core/src/main/java/co/rsk/pcc/bto/BTOUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.bto;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.pcc.NativeContract;
+import co.rsk.pcc.NativeMethod;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Precompiled contract that provides certain BTO (Bitcoin Token Offering)
+ * related utility functions.
+ *
+ * @author Ariel Mendelzon
+ */
+public class BTOUtils extends NativeContract {
+    private final BTOUtilsHelper helper;
+
+    public BTOUtils(RskSystemProperties config, RskAddress contractAddress) {
```

Comment:
I think we shouldn't call this precompiled contract BTOUtils as its not just for BTO but its a number of methods to manage extended pub keys, derivation paths and addresses.

Reply:
Good point. There's also the matter of whether this stays as a single precompiled or rather as more than one grouping different kinds of functions (key handling vs. bitcoin address related). I'd say let's leave that decision to the product owner/s to make at a later stage if they desire to do so.